### PR TITLE
Docs exec testing: pass as soon as containers run.

### PR DESCRIPTION
**Pull request by merlin-northern**
ChangeLog:none
Signed-off-by: Peter Grzybowski <peter@northern.tech>

### CODE REVIEW

File: 07.Server-installation/03.Production-installation/docs.md [hunk 1]
```diff
@@ -613,10 +613,14 @@ installing an Open Source server, please proceed to
 
 <!-- Verification of Open Source instance -->
 
-<!--AUTOMATION: test=for ((n=0;n<11;n++)); do sleep 4 && test "$(docker ps | grep menderproduction | grep -c -i 'up')" = 14  || ( echo "some containers are not 'Up'" && docker ps && ./run images && ./run logs && exit 1 ); done -->
+<!--AUTOMATION: execute=export EXPECTED_COUNT=14 -->
```
Comment (lluiscampos) [inline 07.Server-installation/03.Production-installation/docs.md:616]: Once we are reworking the test... please let's move these magic numbers to explicit variables at the beginning of the file. Something like `EXPECTED_COUNT_ENTERPRISE` and `EXPECTED_COUNT_OPENSOURCE` will be much better. I suggest we do the same in `01.Upgrading-from-OS-to-Enterprise` file, even though there we would only use one of them (enterprise)

File: 07.Server-installation/03.Production-installation/docs.md [hunk 2]
```diff
@@ -613,10 +613,14 @@ installing an Open Source server, please proceed to
...
+<!--AUTOMATION: execute=export SLEEP_INTERVAL=4 -->
+<!--AUTOMATION: execute=export MAX_INTERATIONS=11 -->
+<!--AUTOMATION: execute=function CONTAINERS_COUNT_TEST() { local n; [ "${EXPECTED_COUNT}" == "" ] && return 1; for ((n=0;n<${MAX_INTERATIONS:-11};n++)); do count=$(docker container ls -f "status=running" --format 'table {{.Status}}\t{{.Names}}' | grep ^Up | grep -c menderproduction); [ ${count} -eq ${EXPECTED_COUNT} ] && break; sleep ${SLEEP_INTERVAL:-4}; done; [ ${count} -eq ${EXPECTED_COUNT} ] || { echo "some containers are not 'Up'; ${count}/${EXPECTED_COUNT} running." && docker ps && ./run images && ./run logs && exit 1; }; } -->
```
Comment (lluiscampos) [inline 07.Server-installation/03.Production-installation/docs.md:619]: I don't think the variables here should have defaults as we should expect all to have been defined
```suggestion
<!--AUTOMATION: execute=function CONTAINERS_COUNT_TEST() { local n; [ "${EXPECTED_COUNT}" == "" ] && return 1; for ((n=0;n<${MAX_INTERATIONS};n++)); do count=$(docker container ls -f "status=running" --format 'table {{.Status}}\t{{.Names}}' | grep ^Up | grep -c menderproduction); [ ${count} -eq ${EXPECTED_COUNT} ] && break; sleep ${SLEEP_INTERVAL}; done; [ ${count} -eq ${EXPECTED_COUNT} ] || { echo "some containers are not 'Up'; ${count}/${EXPECTED_COUNT} running." && docker ps && ./run images && ./run logs && exit 1; }; } -->
```

I would also suggest (weak suggestion) to remove the `[ "${EXPECTED_COUNT}" == "" ] && return 1` check, what is the purpose of that?

**Review by lluiscampos — CHANGES_REQUESTED** [review]
Great initiative, thank you!

One request: to please define the magic numbers at the beginning of the files for easier maintenance in the future. What do you think?

File: 07.Server-installation/03.Production-installation/docs.md (hunk 2 above)
Comment (merlin-northern) [reply 07.Server-installation/03.Production-installation/docs.md]: fixed!

> I would also suggest (weak suggestion) to remove the [ "${EXPECTED_COUNT}" == "" ] && return 1 check, what is the purpose of that?

to fail the test if the count is not set? I can remove, no worries, it made sense to me to have it.

File: 07.Server-installation/03.Production-installation/docs.md (hunk 1 above)
Comment (merlin-northern) [reply 07.Server-installation/03.Production-installation/docs.md]: sure.
moved and renamed, also renamed the function name to indicate the ENT/OS test scenario.

**Review by lluiscampos — APPROVED** [review]

**Pull request merged**